internal/config/client.go

```diff
@@ -236,6 +236,31 @@ func (c *MongoDBClient) GetRealmClient(ctx context.Context) (*realm.Client, erro
 	return realmClient, nil
 }
 
+func (c *MongoDBClient) UntypedAPICall(ctx context.Context, versionHeader, relativePath string, pathParams map[string]string, method string, body []byte) (*http.Response, error) {
```

Comment:
The signature of `pathParams` should be `pathParams map[string]string` instead of `pathParams map[string][]string`. There are endpoints which allow multiple parameter values for the same parameter.

Some example operations:
- `listCloudProviderRegions`, parameter: `providers`
- `getCollStatsLatencyNamespaceClusterMeasurements`, parameter: `metrics`
- `listOrganizationEvents`, parameter: `eventType`

Reply:
Thanks for sharing these examples! I do see that all of these cases are query parameters. For the scope of our current PoC we will not require supporting query params so for now I prefer to not include this, you will see that when calling PrepareRequest it is left as `nil`.